### PR TITLE
Add TestResourceReconciliationIdempotency for usercluster

### LINKED ISSUE #2342: Test idempotency of controllers that create kubernetes objects
**User Story**

As a developer I want my CI system to automatically test as much as possible and give me meaningful feedback. 

**Acceptance criteria**

* There is a test that verifies that the controllers won't update any resources after they got created

**Notes**

May be a good usecase for kind: https://github.com/kubernetes-sigs/kind

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Verifies the idempotency of the resource deployment in the usercluster controller.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2342 

Needs https://github.com/kubermatic/infra/pull/799 so we have a current version of the kube-apiserver.

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: api/pkg/controller/user-cluster-controller-manager/resources/reconciler_test.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
+package usercluster
```
Comment (kron4eg) [inline api/pkg/controller/user-cluster-controller-manager/resources/reconciler_test.go:3]: I think you forgot a build tag (integration AFAIK)

**Review by kron4eg — APPROVED** [review]
/lgtm
/approve

File: api/pkg/controller/user-cluster-controller-manager/resources/reconciler_test.go (hunk 1 above)
Comment (alvaroaleman) [reply api/pkg/controller/user-cluster-controller-manager/resources/reconciler_test.go:3]: Fixed

**Pull request merged**